src/main/webapp/go.js

```diff
@@ -436,6 +445,44 @@ function checkUserDestinationGuess() { //eslint-disable-line
   });
 }
 
+/**
+ * Retrieves the place ID of the current destination, retrieved
+ * using the Places library.
+ */
+function getPlaceIDOfAnswer() {
+  const answerRequest = {
+    query: hunt.getCurDestName(),
+    fields: ['place_id'],
+  };
+  service.findPlaceFromQuery(answerRequest, function(results, status) {
```

Comment:
this is an async function so you should await the result

Reply:
This function doesn't return a Promise though, so the current callback function structure seems the most natural to me.

Reply:
In that case I think the callback make the second call to findPlaceFromQuery.  I'm thinking that since these APIs have to make network requests to get the result that you should make sure you have values to evaluate.